src/main/java/teammates/ui/webapi/GetNotificationsAction.java

```diff
@@ -42,15 +45,12 @@ void checkSpecificAccessControl() throws UnauthorizedAccessException {
     @Override
     public JsonResult execute() {
         String targetUserString = getRequestParamValue(Const.ParamsNames.NOTIFICATION_TARGET_USER);
-        // TODO: Use isFetchingAll to decide whether to fetch unread notification only.
-        // boolean isFetchingAll = Boolean.parseBoolean(
-        //     getRequestParamValue(Const.ParamsNames.NOTIFICATION_IS_FETCHING_ALL));
-
         List<NotificationAttributes> notificationAttributes;
         if (targetUserString == null && userInfo.isAdmin) {
-            // if the admin wants to retrieve all notifications
+            // if admin does not specify targetUser, retrieve all notifications
```

Comment:
I believe the admin is not able to specify target user?

Reply:
Admin might still be an instructor/student, adding a check for target user will make sure that user will get all notifications when he does not specify target user. Other than this situation, the API will return the specified target user notifications or throw error when proper.

Reply:
From what our team discussed, we assume admin is accessing instructor/student notifs page when admin specifies user type.

Reply:
I see! Thanks for the clarification, perhaps a rephrase would be clearer: `// if request is from admin and targetUser is not specified, retrieve all notifications`

Reply:
Okay thank you!